packages/element-web-module-api/src/api/custom-components.ts

```diff
@@ -64,6 +64,15 @@ export interface CustomComponentsApi {
      *
      * @param eventType - The event type this renderer is for. Use a RegExp instance if you want to target multiple types.
      * @param renderer - The render function.
+     * @example
+     * ```
+     *  customComponents.registerMessageRenderer("m.room.message", (props, originalComponent) => {
+     *       return <YourCustomComponent mxEvent={props.mxEvent} />;
+     *  });
+     *  customComponents.registerMessageRenderer(/m\.room\.(topic|name)/, (props, originalComponent) => {
+     *       return <YourCustomStateRenderer mxEvent={props.mxEvent} />;
+     *  });
```

Comment:
Shouldn't the API distinct state vs not rather than being quite foot-gunny such that if the latter sees an `m.room.name` timeline event it'd render it?

Reply:
yeah, we could tighten up this example.

Reply:
I'm not sure what we can do here for the short term, to be honest. I feel like trying to factor `MatrixEvent` into some sort of common structure is going to have a slight performance overhead per event, and it just feels gross to maintain another structure that we'd have to keep in line with changes to the event structure. 

I don't know about the rest of product here but I'd be more in favor of just having a breaking change when the rust-sdk pieces land to support whatever comes next.